internal/middleware/logging.go

```diff
@@ -2,37 +2,76 @@ package middleware
 
 import (
 	"context"
+	"encoding/json"
 	"time"
 
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
 	"github.com/openfga/openfga/pkg/logger"
 	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
 )
 
+const (
+	methodKey          = "method"
+	requestIDKey       = "request_id"
+	traceIDKey         = "trace_id"
+	requestDurationKey = "request_duration"
+	tookKey            = "took" // TODO: remove in the future
```

Comment:
This can be dropped all together and we can adjust either the log pipeline or the ingestion pipeline.